packages/lake/src/components/SegmentedControl.tsx

```diff
@@ -34,171 +63,57 @@ export const SegmentedControl = <T extends string>({
-      direction="row"
-      style={{
-        backgroundColor: colors.gray[50],
-        height: HEIGHT,
-        borderRadius: radii[8],
-      }}
-    >
-      <Box
-        role="none"
-        aria-hidden={true}
-        direction="row"
-        style={{
-          position: "absolute",
-          visibility: "hidden",
-        }}
-      >
-        {itemsToDisplay.map((item, index) => {
-          return (
-            <Box
-              key={`${item.id}-hidden`}
-              direction="row"
-              alignItems="center"
-              onLayout={({ nativeEvent: { layout } }) => {
-                itemWidths.current[index] = layout.width;
-              }}
-              style={{
-                padding: spacings[12],
-                backgroundColor: "green",
-              }}
-            >
-              {isNotNullish(item.icon) && (
-                <>
-                  {item.icon}
-
-                  <Space width={12} />
-                </>
-              )}
-
-              <LakeText
-                color={colors.gray[900]}
-                numberOfLines={1}
-                style={{
-                  userSelect: "none",
-                }}
-              >
-                {item.name}
-              </LakeText>
-            </Box>
-          );
-        })}
-      </Box>
-
-      <View
-        style={{
-          flexGrow: 1,
-          flexShrink: 1,
-          padding: spacings[4],
-        }}
-      >
-        <Box direction="row" style={commonStyles.fill}>
-          <View
-            role="none"
-            style={{
-              position: "absolute",
-              left: 0,
-              top: 0,
+    <Box style={styles.container}>
+      <Box direction={mode === "mobile" ? "column" : "row"}>
+        <View
+          role="none"
+          style={[
+            styles.selectedItem,
+            // eslint-disable-next-line react-native/no-inline-styles
+            mode === "mobile" && {
+              right: 0,
+              height: `${(1 / items.length) * 100}%`,
+              transform: `translateY(${selectedItemIndex * 100}%)`,
+            },
+            // eslint-disable-next-line react-native/no-inline-styles
+            mode === "desktop" && {
               bottom: 0,
-              flexGrow: 1,
-              flexShrink: 1,
-              width: `${(1 / itemsToDisplay.length) * 100}%`,
-              transitionProperty: "transform",
-              transitionDuration: "250ms",
-              transitionTimingFunction: "ease",
+              width: `${(1 / items.length) * 100}%`,
               transform: `translateX(${selectedItemIndex * 100}%)`,
-            }}
-            onLayout={({ nativeEvent: { layout } }) => {
-              indicatorWidth.current = layout.width;
-              maybeHideItems();
+            },
+          ]}
+        />
+
+        {items.map(item => (
+          <Pressable
+            key={item.id}
+            style={[styles.item, mode === "desktop" && styles.itemMobile]}
```

Comment:
`mode === "desktop" && styles.itemMobile` -> I think you want to name that `styles.itemDesktop`